### PR TITLE
[DNM] zephyr/main: support for trigger recovery DFU by the application

### PR DESCRIPTION
**Pull request by nvlsianpu**
requires https://github.com/zephyrproject-rtos/zephyr/pull/44512
**DNM** before  ^^

This Patch introduces possibility to trigger the serial recovery mode by the application using dedicated zephyr-rtos API.
Additionally Kconfig and USB DFU class invoking was refactored.

### CODE REVIEW

File: boot/zephyr/Kconfig.serial_recovery [hunk 1]
```diff
@@ -56,6 +56,12 @@ config BOOT_MAX_LINE_INPUT_LEN
 	help
 	  Maximum length of commands transported over the serial port.
 
+config BOOT_SERIAL_USE_PIN_DETECT
```
Comment (nvlsianpu) [inline boot/zephyr/Kconfig.serial_recovery:59]: worth to add an option for "exclusively detect whether the application requested the DFU mode"

File: boot/zephyr/Kconfig [hunk 2]
```diff
@@ -376,81 +376,7 @@ config BOOT_FIH_PROFILE_HIGH
...
-	select IMG_MANAGER
-	select BOOT_USB_DFU
-	help
-	  If y, MCUboot waits for a prescribed duration of time to allow
-	  for USB DFU to be invoked. Please note DFU always updates the
-	  slot1 image.
-
-config BOOT_USB_DFU_GPIO
-	bool "Use GPIO to detect whether to trigger DFU mode"
-	select USB_DEVICE_STACK
```
Comment (de-nordic) [inline boot/zephyr/Kconfig:398]: I advise you to do this as separate PR after https://github.com/mcu-tools/mcuboot/pull/1344 gets in, conflicts with moved files are not fun.

**Comment by carlescufi** [comment]
@anangl please take a look
